x/incentives/keeper/gauge_asset.go

```diff
@@ -0,0 +1,191 @@
+package keeper
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/dymensionxyz/dymension/v3/x/incentives/types"
+	lockuptypes "github.com/osmosis-labs/osmosis/v15/x/lockup/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// distributionInfo stores all of the information for pent up sends for rewards distributions.
+// This enables us to lower the number of events and calls to back.
+type distributionInfo struct {
+	nextID            int
+	lockOwnerAddrToID map[string]int
+	idToBech32Addr    []string
+	idToDecodedAddr   []sdk.AccAddress
+	idToDistrCoins    []sdk.Coins
+	// TODO: add totalDistrCoins to track total coins distributed
+}
+
+// newDistributionInfo creates a new distributionInfo struct
+func newDistributionInfo() distributionInfo {
+	return distributionInfo{
+		nextID:            0,
+		lockOwnerAddrToID: make(map[string]int),
+		idToBech32Addr:    []string{},
+		idToDecodedAddr:   []sdk.AccAddress{},
+		idToDistrCoins:    []sdk.Coins{},
+	}
+}
+
+// getLocksToDistributionWithMaxDuration returns locks that match the provided lockuptypes QueryCondition,
+// are greater than the provided minDuration, AND have yet to be distributed to.
+func (k Keeper) getLocksToDistributionWithMaxDuration(ctx sdk.Context, distrTo lockuptypes.QueryCondition, minDuration time.Duration) []lockuptypes.PeriodLock {
+	switch distrTo.LockQueryType {
+	case lockuptypes.ByDuration:
+		denom := distrTo.Denom
+		if distrTo.Duration > minDuration {
+			return k.lk.GetLocksLongerThanDurationDenom(ctx, denom, minDuration)
+		}
+		return k.lk.GetLocksLongerThanDurationDenom(ctx, distrTo.Denom, distrTo.Duration)
+	case lockuptypes.ByTime:
+		panic("Gauge by time is present, however is no longer supported. This should have been blocked in ValidateBasic")
+	default:
+	}
+	return []lockuptypes.PeriodLock{}
```

Comment:
nil

Reply:
it's better this way as the callers doesn't check for != nil

Reply:
it's better this way as the callers doesn't check for != nil